tools/inc/stljobs.h

```diff
@@ -0,0 +1,606 @@
+        if (hImpl != EmptyPolicy::Empty) {
+            CloseHandle(hImpl);
+            hImpl = EmptyPolicy::Empty;
+        }
+    }
+
+    [[nodiscard]] explicit operator bool() const noexcept {
+        return hImpl != EmptyPolicy::Empty;
+    }
+
+    [[nodiscard]] HANDLE get() const noexcept {
+        return hImpl;
+    }
+
+    [[nodiscard]] HANDLE detach() noexcept {
+        return std::exchange(hImpl, EmptyPolicy::Empty);
+    }
+
+    [[nodiscard]] HANDLE* out() noexcept {
+        return &hImpl;
+    }
+
+private:
+    HANDLE hImpl{EmptyPolicy::Empty};
+};
+
+inline handle<invalid_handle_value_policy> create_file(const wchar_t* const lpFileName, unsigned long dwDesiredAccess,
+    unsigned long dwShareMode, LPSECURITY_ATTRIBUTES lpSecurityAttributes, unsigned long dwCreationDisposition,
+    unsigned long dwFlagsAndAttributes, HANDLE hTemplateFile) {
+    handle<invalid_handle_value_policy> result{CreateFileW(lpFileName, dwDesiredAccess, dwShareMode,
+        lpSecurityAttributes, dwCreationDisposition, dwFlagsAndAttributes, hTemplateFile)};
+    if (!result) {
+        api_failure("CreateFileW");
+    }
+
+    return result;
+}
+
+inline handle<invalid_handle_value_policy> create_named_pipe(LPCWSTR lpName, DWORD dwOpenMode, DWORD dwPipeMode,
+    DWORD nMaxInstances, DWORD nOutBufferSize, DWORD nInBufferSize, DWORD nDefaultTimeOut,
+    LPSECURITY_ATTRIBUTES lpSecurityAttributes) {
+    handle<invalid_handle_value_policy> result{CreateNamedPipeW(lpName, dwOpenMode, dwPipeMode, nMaxInstances,
+        nOutBufferSize, nInBufferSize, nDefaultTimeOut, lpSecurityAttributes)};
+    if (!result) {
+        api_failure("CreateNamedPipeW");
+    }
+
+    return result;
+}
+
+
+const auto is_space = [](const wchar_t c) { return c == L' '; };
+
+[[nodiscard]] inline handle<null_handle_policy> create_kill_on_job_close_job() {
+    handle<null_handle_policy> hJob{CreateJobObjectW(nullptr, nullptr)};
+    if (!hJob) {
+        api_failure("CreateJobObjectW");
+    }
+
+    JOBOBJECT_EXTENDED_LIMIT_INFORMATION limits{};
+    limits.BasicLimitInformation.LimitFlags =
+        JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE | JOB_OBJECT_LIMIT_DIE_ON_UNHANDLED_EXCEPTION;
+    if (!SetInformationJobObject(hJob.get(), JobObjectExtendedLimitInformation, &limits, sizeof(limits))) {
+        api_failure("SetInformationJobObject");
+    }
+
+    return hJob;
+}
+
+inline void put_self_in_job() {
+    auto hJob = create_kill_on_job_close_job();
+
+    // Put ourselves in that job
+    if (!AssignProcessToJobObject(hJob.get(), GetCurrentProcess())) {
+        api_failure("AssignProcessToJobObject");
+    }
+
+    // Purposely leak hJob to avoid terminating ourselves
+    (void) hJob.detach();
+}
+
+class no_input_pipe {
+public:
+    no_input_pipe() {
+        SECURITY_ATTRIBUTES inheritSa;
+        inheritSa.nLength              = sizeof(inheritSa);
+        inheritSa.lpSecurityDescriptor = nullptr;
+        inheritSa.bInheritHandle       = TRUE;
+        HANDLE read;

```

Comment:
Some match Win32 conventions exactly because they are being mapped into that universe which uses hungarian notation, and some are entirely of this code's design and do not.

`hDevNull` shouldn't have the h.